proto/public/uber/cadence/api/v1/service_workflow.proto

```diff
@@ -0,0 +1,255 @@
+
+  // TerminateWorkflowExecution terminates an existing workflow execution by recording WorkflowExecutionTerminated event
+  // in the history and immediately terminating the execution instance.
+  rpc TerminateWorkflowExecution(TerminateWorkflowExecutionRequest) returns (TerminateWorkflowExecutionResponse);
+
+  // DescribeWorkflowExecution returns information about the specified workflow execution.
+  rpc DescribeWorkflowExecution(DescribeWorkflowExecutionRequest) returns (DescribeWorkflowExecutionResponse);
+
+  // QueryWorkflow returns query result for a specified workflow execution.
+  rpc QueryWorkflow(QueryWorkflowRequest) returns (QueryWorkflowResponse);
+
+  // DescribeTaskList returns information about the target tasklist, right now this API returns the
+  // pollers which polled this tasklist in last few minutes.
+  rpc DescribeTaskList(DescribeTaskListRequest) returns (DescribeTaskListResponse);
+
+  // ListTaskListPartitions returns information about task list partitions.
+  rpc ListTaskListPartitions(ListTaskListPartitionsRequest) returns(ListTaskListPartitionsResponse);
+
+  // Reset the sticky tasklist related information in mutable state of a given workflow.
+  // Things cleared are:
+  // 1. StickyTaskList
+  // 2. StickyScheduleToStartTimeout
+  // 3. ClientLibraryVersion
+  // 4. ClientFeatureVersion
+  // 5. ClientImpl
+  rpc ResetStickyTaskList(ResetStickyTaskListRequest) returns(ResetStickyTaskListResponse);
+
+  // GetClusterInfo returns information about cadence cluster.
+  rpc GetClusterInfo(GetClusterInfoRequest) returns(GetClusterInfoResponse);
+
+  // Returns the history of specified workflow execution.  It fails with 'EntityNotExistError' if specified workflow
+  // execution in unknown to the service.
+  rpc GetWorkflowExecutionHistory(GetWorkflowExecutionHistoryRequest) returns(GetWorkflowExecutionHistoryResponse);
+}
+
+
+message StartWorkflowExecutionRequest {
+  string domain = 1;
+  string workflow_id = 2;
+  WorkflowType workflow_type = 3;
+  TaskList task_list = 4;
+  Payload input = 5;
+  google.protobuf.Duration execution_start_to_close_timeout = 6 [(gogoproto.stdduration) = true];
+  google.protobuf.Duration task_start_to_close_timeout = 7 [(gogoproto.stdduration) = true];
+  string identity = 8;
+  string request_id = 9;
+  WorkflowIdReusePolicy workflow_id_reuse_policy = 10;
+  RetryPolicy retry_policy = 11;
+  string cron_schedule = 12;
+  Memo memo = 13;
+  SearchAttributes search_attributes = 14;
+  Header header = 15;
+}
+
+message StartWorkflowExecutionResponse {
+  string run_id = 1;
+}
+
+message SignalWorkflowExecutionRequest {
+  string domain = 1;
+  WorkflowExecution workflow_execution = 2;
+  string signal_name = 3;
+  Payload input = 4;
+  string identity = 5;
+  string request_id = 6;
+  bytes control = 7;
+}
+
+message SignalWorkflowExecutionResponse {
+}
+
+message SignalWithStartWorkflowExecutionRequest {
```

Comment:
may be worth embedding `StartWorkflowExecutionRequest`, and adding `SignalWorkflowExecutionRequest` fields separately?  Or splitting out the unique fields of both to a separate message type.  Since they really are "the same thing" - if we add something to Start, we probably want it available on SignalWithStart too.

Reply:
Updated.